3rdparty/mbedtls/mbedtls/library/x509write_crt.c

```diff
@@ -325,25 +325,25 @@ static int x509_write_time( unsigned char **p, unsigned char *start,
     return( (int) len );
 }
 
-int mbedtls_x509write_crt_der( mbedtls_x509write_cert *ctx, unsigned char *buf, size_t size,
-                       int (*f_rng)(void *, unsigned char *, size_t),
-                       void *p_rng )
+int mbedtls_x509write_crt_der( mbedtls_x509write_cert *ctx,
+                               unsigned char *buf, size_t size,
+                               int (*f_rng)(void *, unsigned char *, size_t),
+                               void *p_rng )
 {
     int ret;
     const char *sig_oid;
     size_t sig_oid_len = 0;
     unsigned char *c, *c2;
     unsigned char hash[64];
     unsigned char sig[MBEDTLS_MPI_MAX_SIZE];
-    unsigned char tmp_buf[2048];
     size_t sub_len = 0, pub_len = 0, sig_and_oid_len = 0, sig_len;
     size_t len = 0;
     mbedtls_pk_type_t pk_alg;
 
     /*
-     * Prepare data to be signed in tmp_buf
+     * Prepare data to be signed at the end of the target buffer
      */
-    c = tmp_buf + sizeof( tmp_buf );
+    c = buf + size;
```

Comment:
Is there plans to upstream this change?

Reply:
Not to my knowledge, so I filed #2179 when I came across this during the update to 2.7.11

Reply:
There was an issue opened by @soccerGB with mbedTLS several months ago about this: https://github.com/ARMmbed/mbedtls/issues/2631

From the comments, it looks like they've merged the changes in successfully, although I'm not sure which release they're available in: https://github.com/ARMmbed/mbedtls/pull/2632